core/src/model.jl

```diff
@@ -97,27 +101,31 @@ function Model(config::Config)::Model
 
         # tell the solver to stop when new data comes in
         tstops = Vector{Float64}[]
-        for schema_version in [
-            DiscreteControlConditionV1,
-            FlowDemandTimeV1,
-            LevelDemandTimeV1,
-            PidControlTimeV1,
-            TabulatedRatingCurveTimeV1,
-            UserDemandTimeV1,
-        ]
+        for schema_version in [PidControlTimeV1, TabulatedRatingCurveTimeV1]
             time_schema = load_structvector(db, config, schema_version)
             push!(tstops, get_tstops(time_schema.time, config.starttime))
         end
 
         # Tell the solver to stop at all data points from periodically extrapolated
         # if applicable, otherwise just the original timeseries
         for interpolations in [
-            basin.forcing.precipitation,
-            basin.forcing.potential_evaporation,
             basin.forcing.drainage,
             basin.forcing.infiltration,
-            level_boundary.level,
+            basin.forcing.potential_evaporation,
+            basin.forcing.precipitation,
             flow_boundary.flow_rate,
+            flow_demand.demand_itp,
+            level_boundary.level,
+            level_demand.max_level,
+            level_demand.min_level,
+            user_demand.demand_itp...,
+            user_demand.return_factor,
+            vcat(
+                [
+                    [cv.greater_than for cv in cvs] for
+                    cvs in discrete_control.compound_variables
+                ]...,
+            )...,
```

Comment:
Would be good to avoid this splatting. From the vcat docs:
```
To concatenate a large vector of arrays, reduce(vcat, A) calls an efficient method when A isa
  AbstractVector{<:AbstractVecOrMat}, rather than working pairwise.
```